### ISSUE TITLE
JIT is not generating vzeroupper when using AVX on unix

### ISSUE THREAD

**Issue by CarolEidt** [dotnet/coreclr#5166]
We are generating the vzeroupper in the `genPreserveCalleeSavedFltRegs` and `genRestoreCalleeSavedFltRegs methods`, but only if we have any callee-save registers, which, of course, we never do on Linux. Those should either be moved before these lines (in both methods):

```
    // fast path return
    if (regMask == RBM_NONE) 
        return;
```

Or somewhere else. This is because the vzeroupper instruction is necessary to avoid penalties in the event that the method is being called from a method that uses SSE encodings.

**Comment by RussKeldorph** [comment]
Need to assess perf impact ASAP.

**Comment by CarolEidt** [comment]
@erozenfeld benchmarked this, and the difference was in the noise. I think that this needs further analysis before making the change, since our current (Windows) implementation only issues it when the jitted code is the callee, but not when it is calling into possibly-SSE code (e.g. if the target is precompiled or native). So I believe the right thing to do is to defer this for now.

**Comment by CarolEidt** [comment]
@RussKeldorph do you agree that deferring this is appropriate?

**Comment by RussKeldorph** [comment]
@CarolEidt Sounds good to me.  Moving out.  Thanks.

**Comment by sivarv** [comment]
Intel analysis of TechEmPower indicates SSE to AVX transition penalty is incurred in Kestral server dll (Microsoft.AspNetCore.Server.Kestrel.dll module on Ubuntu/Windows) both on Ubuntu and Windows.

Right now on Unix we don't emit Vzeroupper instruction.  We need to see whether emitting would mitigate some of those transitions.

In addition to that what the analysis point is that when jitted could potentially calls to SSE code (e.g. PInvoke), there is no Vzeroupper issued right now.

**Comment by CarolEidt** [comment]
This was fixed with #8588
